src/panel/pages/timeline/components/TimelineTooltip.tsx

```diff
@@ -42,59 +55,103 @@ const TooltipElement = styled.p<{ position?: TooltipPosition }>`
     border-left: 6px solid transparent;
     border-right: 6px solid transparent;
     margin-top: -1px;
-    left: 50%;
+    left: calc(50% - ${(p) => p.offset}px);
     top: 100%;
     transform: translate(-50%, 0);
   }
 `;
 
 export const useTooltip = () => {
-  const ref: MutableRefObject<HTMLElement | null> = useRef<HTMLElement>(null);
+  const containerRef: MutableRefObject<HTMLElement | null> = useRef<
+    HTMLElement
+  >(null);
+  const tooltipRef: MutableRefObject<HTMLElement | null> = useRef<HTMLElement>(
+    null
+  );
+  const tooltipOffset: MutableRefObject<number> = useRef(0);
+  const mouseX = useRef<number | undefined>(undefined);
   const [hasRef, setHasRef] = useState(false);
   const [isVisible, setIsVisible] = useState(false);
   const [tooltipProps, setTooltipProps] = useState<
-    ComponentProps<typeof TimelineTooltip>
+    ComponentProps<typeof TimelineTooltip> & TimelineTooltipProps
   >({});
 
   const calculateTooltipPosition = useCallback(() => {
-    if (!ref.current) {
+    if (!containerRef.current) {
       return;
     }
 
-    const { x, y, width } = ref.current.getBoundingClientRect(); //eslint-disable-line @typescript-eslint/no-non-null-assertion
-    setTooltipProps({
+    const { x, y, width } = containerRef.current.getBoundingClientRect(); //eslint-disable-line @typescript-eslint/no-non-null-assertion
```

Comment:
Looking at your prior conditional, it seems like eslint shouldn't be throwing errors
```suggestion
    const { x, y, width } = containerRef.current.getBoundingClientRect();
```